Dfe.Academies.Academisation.Service/Commands/Legacy/Project/LegacyProjectUpdateCommand.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Dfe.Academies.Academisation.Core;
+using Dfe.Academies.Academisation.IData.ProjectAggregate;
+using Dfe.Academies.Academisation.IService.Commands.Project;
+using Dfe.Academies.Academisation.IService.ServiceModels.Legacy.ProjectAggregate;
+using Dfe.Academies.Academisation.Service.Mappers.Legacy.ProjectAggregate;
+
+namespace Dfe.Academies.Academisation.Service.Commands.Project;
+
+public class LegacyProjectUpdateCommand : ILegacyProjectUpdateCommand
+{
+	private readonly IProjectGetDataQuery _projectGetDataQuery;
+	private readonly IProjectUpdateDataCommand _projectUpdateDataCommand;
+
+	public LegacyProjectUpdateCommand(IProjectGetDataQuery projectGetDataQuery, IProjectUpdateDataCommand projectUpdateDataCommand)
+	{
+		_projectGetDataQuery = projectGetDataQuery;
+		_projectUpdateDataCommand = projectUpdateDataCommand;
+	}
+
+	public async Task<CommandResult> Execute(LegacyProjectServiceModel legacyProjectServiceModel)
+	{
+		var existingProject = await _projectGetDataQuery.Execute(legacyProjectServiceModel.Id);
+
+		if (existingProject is null)
+		{
+			return new NotFoundCommandResult();
+		}
+
```

Comment:
Could put the patch logic here to create a merged ProjectDetails to pass in as the parameter to an Update method. It also fits, because the patch approach feels a bit like it belongs in a legacy command.